libraries/AP_HAL_ChibiOS/hwdef/SIYI_N7/hwdef.dat

```diff
@@ -159,15 +159,26 @@ PB10 nSPI5_RESET_EXTERNAL OUTPUT HIGH
 # SPI devices
 SPIDEV ms5611         SPI4 DEVID1  MS5611_CS    MODE3 20*MHZ 20*MHZ
 SPIDEV icm20689       SPI1 DEVID1  ICM20689_CS  MODE3  2*MHZ  8*MHZ
+SPIDEV icm45686       SPI1 DEVID1  ICM20689_CS  MODE3  2*MHZ  8*MHZ
 SPIDEV bmi088_g       SPI1 DEVID3  BMI088_G_CS  MODE3 10*MHZ 10*MHZ
 SPIDEV bmi088_a       SPI1 DEVID4  BMI088_A_CS  MODE3 10*MHZ 10*MHZ
 SPIDEV ramtron        SPI2 DEVID1  FRAM_CS      MODE3  8*MHZ  8*MHZ 
 
 # Two IMUs
 IMU Invensense SPI:icm20689 ROTATION_NONE
+IMU Invensensev3 SPI:icm45686 ROTATION_ROLL_180_YAW_270  
 IMU BMI088 SPI:bmi088_a SPI:bmi088_g ROTATION_ROLL_180_YAW_90
 
 define HAL_DEFAULT_INS_FAST_SAMPLE 3
+define ICM45686_CLKIN 1
+define HAL_INS_HIGHRES_SAMPLE 3
+
+CHECK_ICM20689        spi_check_register_inv2("icm20689", INV2REG_WHOAMI, INV2_WHOAMI_ICM20689)
+CHECK_ICM45686        spi_check_register("icm45686", INV3REG_456_WHOAMI, INV3_WHOAMI_ICM45686)
+
+CHECK_IMU0_PRESENT      $CHECK_ICM20689 || $CHECK_ICM45686
+
+BOARD_VALIDATE  $CHECK_IMU0_PRESENT 
```

Comment:
Any particular reason for adding the validation code?

The board's not going to accomplish much without an IMU!

The validation stuff is really meant for multi-IMU boards